### PR TITLE
Use ci-server as DNS instead of /etc/hosts entries

### PR DESCRIPTION
**Pull request by tervis-unikie**
Also updated copyrights of the relevant files and added ficolo DNS setup also to prbuilder.

Tested on prbuilder.

### CODE REVIEW

**Review by mkaapu — APPROVED** [review]

**Review by karim20230 — APPROVED** [review]

File: hosts/default.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@
     # shared modules
     azure-common = import ./azure-common.nix;
     qemu-common = import ./qemu-common.nix;
-    ficolo-hosts = import ./ficolo-hosts.nix;
+    ficolo-common = import ./ficolo-common.nix;
```
Comment (cazfi) [inline hosts/default.nix:14]: Not a new issue with this PR, but it seems wrong to me that ficolo-specific (even if common to all ficolo servers) configuration is in hosts/default.nix common to all servers.

**Review by mnokka — APPROVED** [review]
Logical way to change DNS

File: hosts/default.nix (hunk 1 above)
Comment (cazfi) [reply hosts/default.nix:14]: Thinking more about it, what used to be "wrong" turns to "breaking things" with this PR. We don't want to set those dns servers for machines outside ficolo.

Comment (tervis-unikie) [reply hosts/default.nix:14]: Afaik this just "declares" the 'ficolo-common' as importing ficolo-common.nix.
As long as e.g. azure machines do not use 'ficolo-common' at all, they won't include any ficolo stuff into configuration.
And if this breaks things, it was broken before this PR.

**Review by cazfi — APPROVED** [review]

File: hosts/default.nix (hunk 1 above)
Comment (mkaapu) [reply hosts/default.nix:14]: Yes. This is how @tervis-unikie put it. This just 'introduces' or 'declares' a new module. You still need to include it separately to take it into use with any specific configuration.

**Review by mkaapu — APPROVED** [review]

**Pull request merged**